src/nanorc/sshpm.py

```diff
@@ -345,7 +372,21 @@ def terminate(self):
                 except OSError:
                     pass
         self.apps = {}
-
+        for name, desc in self.services.items():
+            if desc.proc is not None and desc.proc.is_alive():
+                try:
+                    desc.proc.terminate()
+                    while desc.proc.is_alive():
+                        time.sleep(0.1)
+                except OSError:
+                    pass
+            pid_file = f"{name}.pid"
```

Comment:
What does this pid_file do?

Reply:
The gunicorn server will write its pid to the pid file. Then when we want to stop it we can send a terminate signal to that pid. Other servers may also work that way, many linux services use pid files in startup/stop. Killing the ssh that launched the service will not kill the service because it detaches itself from the ssh session.